src/Shared/AssemblyUtilities.cs

```diff
@@ -65,12 +65,19 @@ public static AssemblyName CloneIfPossible(this AssemblyName assemblyNameToClone
             name.SetPublicKey(assemblyNameToClone.GetPublicKey());
             name.SetPublicKeyToken(assemblyNameToClone.GetPublicKeyToken());
             name.Version = assemblyNameToClone.Version;
+            name.Flags = assemblyNameToClone.Flags;
+            name.CultureName = assemblyNameToClone.CultureName;
```

Comment:
Setting `CultureName` will throw NIE on mono. You can set `AssemblyName.CultureInfo` instead. See https://github.com/mono/msbuild/commit/d4c871554368485fde1dbaed09922319dd7f4ea7 and https://github.com/mono/msbuild/commit/42ba59a920754c24968dafe82622a59d27385c1d .

Reply:
@radical could you please file a bugzilla so we get this implemented?

Reply:
Assigning it when `!MONO`. CultureInfo gets set via `FEATURE_ASSEMBLYNAME_CULTUREINFO`